platform/security/idp/security-idp-client/src/main/java/org/codice/ddf/security/idp/client/IdpHandler.java

```diff
@@ -245,6 +257,51 @@ public Object getAttribute(String name) {
             return doPaosRequest(request, response);
         }
 
+        if (userAgentCheck && userAgentIsNotBrowser(httpRequest)) {
+            SecurityLogger.audit("Attempting to log client in as a legacy system.");
+            //if we get here, it is most likely an older DDF that is federating
+            //it isn't going to understand the redirect to the IdP and it doesn't support ECP
+            //so we need to fall back to other handlers to allow it to log in using PKI, Basic or Guest
+
+            //check for basic auth first
+            BasicAuthenticationHandler basicAuthenticationHandler = new BasicAuthenticationHandler();
```

Comment:
Yeah, I'm mulling over Brendan's comment. If I were to go the direction of using any additionally configured handlers, then it would automatically allow you to do this.